zerver/lib/cache.py

```diff
@@ -672,13 +668,6 @@ def flush_stream(
     from zerver.models import UserProfile
 
     stream = instance
-    items_for_remote_cache = {}
-
-    if update_fields is None:
-        cache_delete(get_stream_cache_key(stream.name, stream.realm_id))
-    else:
-        items_for_remote_cache[get_stream_cache_key(stream.name, stream.realm_id)] = (stream,)
-        cache_set_many(items_for_remote_cache)
 
     if (
         update_fields is None
```

Comment:
Hmm, this next cache flushing block probably could be removed by making `default_sending_stream` and `default_events_register_stream` not included in the `select_related()` calls for `UserProfile` for populating the bot dict cache; it's definitely not necessary, since it's totally fine to just fetch the Stream object by ID in the one place in the send message code path where we use it.

Reply:
We are definitely on the same page here!  I noticed that chunk of code too.  Keep this in the back of your mind.  I intend to address `UserProfile` some time soon, but my brain is too full to tackle this yet.